src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -109,22 +109,40 @@ struct CPUStreamsExecutor::Impl {
             auto stream_processors = _impl->_config.get_stream_processor_ids();
             _numaNodeId = numa_node_id;
             _socketId = socket_id;
+            int tbb_version;
+#    if (TBB_INTERFACE_VERSION < 12000)
+            tbb_version = tbb::TBB_runtime_interface_version();
+#    else
+            tbb_version = TBB_runtime_interface_version();
+#    endif
             if (stream_type == STREAM_WITHOUT_PARAM) {
-                _taskArena.reset(new custom::task_arena{custom::task_arena::constraints{}
-                                                            .set_max_concurrency(concurrency)
-                                                            .set_max_threads_per_core(max_threads_per_core)});
+                int version_major = tbb_version / 10;
+                int version_patch = tbb_version % 10;
+                bool support_core_types = false;
+                if ((version_major == TBB_VERSION_MAJOR_CORE_TYPES_WINDOWS &&
+                     version_patch >= TBB_VERSION_PATCH_CORE_TYPES_WINDOWS) ||
+                    (version_major == TBB_VERSION_MAJOR_CORE_TYPES_LINUX &&
+                     version_patch >= TBB_VERSION_PATCH_CORE_TYPES_LINUX)) {
+                    support_core_types = true;
+                }
+                auto core_types = tbb::info::core_types();
```

Comment:
Shouldn't we put this call under `TBB_INTERFACE_VERSION ` condition check macro?

Reply:
Changed `tbb::info::core_types()` to `custom::info::core_types()`, so shouldn't put it under TBB macro check.